### PR TITLE
Host local storage support, enhancements

### PR DESCRIPTION
**Pull request by torchiaf**
#### PR Checklist
- Is this a multi-tenancy feature/bug?
    - [ ] Yes, the relevant RBAC changes are at:
- Do we need to backport changes to the [old Rancher UI](https://github.com/rancher/u), such as RKE1?
    - [ ] Yes, the relevant PR is at:
- Are backend engineers aware of UI changes?
    - [ ] Yes, the backend owner is:

Related Issue https://github.com/harvester/harvester/issues/6685

### Technical notes summary

- Removing 'Export to Image' action for Longhorn V2 volumes.
- Disable 'Size' inputBox for Longhorn V2 volumes in edit mode and add a warning banner - both Volumes and Virtual Machines page.
- Disable 'Volume Encryption' section in Storage Class create page.
- TBD if we want to clean up removed disks from history.

### Screenshot/Video

Volume edit

image

VM's Volume edit

image

### CODE REVIEW

File: pkg/harvester/models/harvester/persistentvolumeclaim.js [hunk 1]
```diff
@@ -30,32 +33,39 @@ export default class HciPv extends HarvesterResource {
...
-    const out = super._availableActions;
+    let out = super._availableActions;
+
     const clone = out.find(action => action.action === 'goToClone');
 
     if (clone) {
       clone.action = 'goToCloneVolume';
     }
 
+    if (this.storageClass.provisioner !== LONGHORN_DRIVER || this.storageClass.longhornVersion !== DATA_ENGINE_V2) {
```
Comment (a110605) [inline pkg/harvester/models/harvester/persistentvolumeclaim.js:44]: Could you revert the change in persistentvolumeclaim.js ? 
we need keep take snapshot for LHv2 volume.

File: pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachineVolume/type/vmImage.vue [hunk 2]
```diff
@@ -112,6 +114,10 @@ export default {
       return allPVCs.find((P) => {
         return this.namespace ? P.id === `${ this.namespace }/${ this.value.volumeName }` : true;
       });
+    },
+
+    isLonghornV2() {
+      return this.value.pvc?.storageClass?.provisioner === LONGHORN_DRIVER && this.value.pvc?.storageClass?.longhornVersion === DATA_ENGINE_V2;
```
Comment (a110605) [inline pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachineVolume/type/vmImage.vue:120]: nit: consider extract `isLonghornV2()` to util function. Maybe put in `utils/storage.js`

**Comment by torchiaf** [comment]
@a110605 I've re-enabled the snapshots action and disable volume envryption for V2 classes + code refactoring, pleate take a look.
`Delete unsaved disk in host edit page should clean related data` point is still open, so I'd keep it as is for the moment.

**Review by tserong — APPROVED** [review]
I've tested locally and can confirm:

- 'Export to Image' is removed for LHv2 volumes
- 'Size' box is disabled and warning appears when editing LHv2 volumes
- Volume encryption is disabled when creating LHv2 storage classes

Thank you!

**Comment by tserong** [comment]
Oh, wait, I just found one little oddity -- the "Longhorn V2 volumes cannot be resized" warning appears when viewing VM detail, but is probably really only necessary on the edit screen:

image

**Comment by torchiaf** [comment]
> the "Longhorn V2 volumes cannot be resized" warning appears when viewing VM detail, but is probably really only necessary on the edit screen
> 
Fixed, thanks.

**Comment by tserong** [comment]
> Fixed, thanks.

Verified!

**Review by tserong — APPROVED** [review]

**Comment by torchiaf** [comment]
@mergify backport release-harvester-v1.4
